### ISSUE TITLE
TextStyle

### ISSUE THREAD

**Issue by vladdeSV** [vladdeSV/scone#51]
Replacement for (or an alternative to) the current way of setting foreground and background colors.

Personally I feel the current approach comes across as clumsy, where you have to write `Color.red.foreground` or `foreground(Color.red)`.

The styles must be easy to use, and where uninitialised properties must remain the same overwritten cells.

I propose a pattern as such

```d
TextStyle().fg(Color.red).bg(Color.green)
```
and
```d
TextStyle(Color.red, Color.green)
```
and
```d
auto style = TextStyle();
style.foreground = Color.red;
style.background = Color.green;
```

**Comment by vladdeSV** [comment]
Sometime soon I would like to add a text-wrapper of sorts. It would consist of a string of text and a text style. This will become it's own GitHub issue later.

```d
Foo("bar", TextStyle().fg(Color.red));
```

**Comment by vladdeSV** [comment]
`Cell` should contain its own `TextStyle`, and not each attribute separately.

Maybe the name TextStyle misleading, as the cell only contains one (1) character (dchar).

**Comment by vladdeSV** [comment]
Currently `enum Color {...}` has all ANSI colors, but also the special colors `initial` and `same`.

ANSI colors actually define a special "default" color, which `initial` represents. However, `same` does not have a ANSI counterpart, and is only used by *scone* in the buffer to use the same color as the cell being overwritten.

With the addition of `TextStyle`, I think I can make the foreground and background colors nullable, and not have a special case in the color enum. All while providing a simple experience using the text style.

---

On second thought, I do not think this is a good idea. Back to the drawing board.

**Comment by vladdeSV** [comment]
(Found out the pattern I proposed is called [*Fluent Interface*](https://martinfowler.com/bliki/FluentInterface.html))

**Comment by vladdeSV** [comment]
Implemented in 9ae99d10903245b1156b57e1048b9648b63c910e

**Comment by vladdeSV** [comment]
Unsure if this should be *the* way to style text. Regardless, the ability to use `Color.red.foreground` should stay in version 3.0 (I think).

Currently it is marked as deprecated. Might undo this later.
